flink-connector/docs/content/docs/connectors/datastream/pubsub.md

```diff
@@ -0,0 +1,158 @@
+without causing messages redelivery. Note that acknowledge deadlines can be
+extended to at most 1h.
+
+### Performance Considerations
+
+- Infrequent checkpointing can cause performance issues, including subscription
+  backlog growth and increased rate of duplicate message delivery.
+- `StreamingPull` connections have a 10 MB/s throughput limit. See
+  [Pub/Sub quotas and limits](https://cloud.google.com/pubsub/quotas) for all
+  restrictions.
+
+<!-- TODO(matt-kwong) Add threading details. -->
+
+### All Options
+
+#### Required Builder Options
+
+<table class="table table-bordered">
+  <thead>
+    <tr>
+      <th class="text-left" style="width: 25%">Builder Method</th>
+      <th class="text-left" style="width: 10%">Default Value</th>
+      <th class="text-left" style="width: 65%">Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+        <td>setSubscriptionName(String subscriptionName)</td>
+        <td>(none)</td>
+        <td>The ID of the subscription from which Pub/Sub source consumes messages.</td>
+    </tr>
+    <tr>
+        <td>setProjectName(String projectName)</td>
+        <td>(none)</td>
+        <td>The ID of the GCP project that owns the subscription from which Pub/Sub source consumes messages.</td>
+    </tr>
+    <tr>
+        <td>setDeserializationSchema(PubSubDeserializationSchema&lt;OutputT&gt; deserializationSchema)</td>
+        <td>(none)</td>
+        <td>How <code>PubsubMessage</code> is deserialized when Pub/Sub source receives a message.</td>
+    </tr>
+  </tbody>
+</table>
+
+#### Optional Builder Options
+
+<table class="table table-bordered">
+  <thead>
+    <tr>
+      <th class="text-left" style="width: 25%">Builder Method</th>
+      <th class="text-left" style="width: 10%">Default Value</th>
+      <th class="text-left" style="width: 65%">Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+        <td>setMaxOutstandingMessagesCount(Long count)</td>
+        <td><code>1000L</code></td>
+        <td>The maximum number of messages that can be delivered to a Pub/Sub source subtask in a checkpoint interval.</td>
+    </tr>
+    <tr>
+        <td>setMaxOutstandingMessagesBytes(Long bytes)</td>
+        <td><code>100L * 1024L * 1024L</code> (100 MB)</td>
+        <td>The maximum number of cumulative bytes that can be delivered to a Pub/Sub source subtask in a checkpoint interval.</td>
+    </tr>
+    <tr>
+        <td>setCredentials(Credentials credentials)</td>
+        <td>(none)</td>
+        <td>The credentials attached to requests sent to Google Cloud Pub/Sub. The identity in the credentials must be authorized to pull messages from the subscription. If not set, then Pub/Sub source attempts to use environment-configured credentials.</td>
+    </tr>
+  </tbody>
+</table>
+
+<!-- TODO(matt-kwong) Add sink details. -->
+
+<!-- TODO(matt-kwong) Add auth details. -->
+
+<!-- TODO(matt-kwong) Add integration testing details. -->
```

Comment:
once submitted, the info in the description of #355 would be helpful to have here

Reply:
Done.